rts/motoko-rts/src/memory/ic.rs

```diff
@@ -28,9 +28,18 @@ extern "C" {
     pub(crate) fn get_static_roots() -> Value;
 }
 
+pub(crate) unsafe fn get_aligned_heap_base() -> u32 {
+    // align to 32 bytes
+    (get_heap_base() + 31) / 32 * 32
+}
+
 #[no_mangle]
-unsafe extern "C" fn init() {
-    HP = get_heap_base() as u32;
+unsafe extern "C" fn init(align: bool) {
+    HP = if align {
+        get_aligned_heap_base()
+    } else {
+        get_heap_base()
+    };
```

Comment:
I wonder if we should align always and keep this code (and the code below) simple. It will waste at most 7 bytes when copying GC is used. Not sure if it's worth making things more complicated. This would simplify tests too.

Reply:
It wastes at most 28 bytes (if we assume that the HP is word-aligned).

Reply:
Somebody relying on being able to use the utmost bit might be tripped over by even such a minor regression. I wouldn't be very happy about receiving such a bug report :-)

Reply:
> Somebody relying on being able to use the utmost bit might be tripped over by even such a minor regression

I don't understand what you mean here. What is "utmost bit"? How can aligning heap base always cause a bug?

Reply:
Oh, if you increase the initial heap pointer then you reduce the available dynamic heap space. You _can_ (but very seldom will) run out of heap space and get a GC earlier than before. Not that it is a sound strategy to rely on it, but some may actually.

Reply:
I doubt anyone will notice that the usable heap size is 27 bytes smaller in some cases, but OK.